thunder/tests/test_inplace_copy.py

```diff
@@ -169,12 +170,21 @@ def func(T0):
     # This pattern is unsafe in general. Disabling sanity check to silence
     # exception for testing
     traced_foo = executor.make_callable(func, disable_inplace_copy_check=True)
-    a = make_tensor((4, 4), device=device, dtype=tdtype)
-    a_ref = a.clone()
-
-    o_thunder = traced_foo(a)
-    o_eager = func(a_ref)
-
-    assert_close(a_ref, a)
-    for o, o_ref in zip(o_thunder, o_eager):
-        assert_close(o, o_ref)
+    t0 = make_tensor((4, 4), device=device, dtype=tdtype)
+    t0_ref = t0.clone()
+
+    actual_t1, actual_t2 = traced_foo(t0)
+
+    expected = t0_ref.sin().cos()
+    expected_t1 = t0_ref.sin()
+    expected_t2 = expected_t1.cos()
+    expected = expected_t2
+
+    assert_close(t0, expected)
+    assert_close(actual_t2, expected_t2)
+    # FIXME(crcrpar): Since there's no `ltorch.Tensor.copy_`, functions like `func` would not
+    # be observed and executed with pytorch eager mode. Though there should be either an audit of
+    # `prims.copy_` in a nvfuser region and/or what #1110 did.
+    assert actual_t1.data_ptr() == actual_t2.data_ptr()
```

Comment:
Thanks for putting a FIXME there. but this is the case that nvfuser is producing wrong result in `func`?

one of the output will still be aliasing the input, and that's not what the program should be translated. i.e. I don't think an audit that removes redundant `prims.copy_` would be able to solve that.

Reply:
For the record, I'm all in for having a `remove redundant inplace copy`, since that's a net improvement. But I'm not sure it would resolve the issue with performance regression that we are seeing right now.

Reply:
If a callable has multiple in-place ops, then the redundant copies would be cleaned up. We just don't have an equivalent for multiple raw `prims.copy_`s